### ISSUE TITLE
aiyprojects - voice - other languages - please advise when available.

### ISSUE THREAD

**Issue by sheridat** [google/aiyprojects-raspbian#253]
Hi, answering questions regarding how to use other languages by saying your working on it https://www.raspberrypi.org/forums/viewtopic.php?f=114&t=200520

Without hassling yourselves - any idea how long it might be.

**Comment by PaulFinch** [comment]
Other languages are already available (i use it), but curiously on the voicekit branch (which use the google assistant 0.1.0 version). I think the aiyprojects branch will be updated soon.

**Comment by sheridat** [comment]
@PaulFinch Hi it is the aiyprojects branch that I am interested in (I've also got it working on voicekit). I've changed the title of this issue to hopefully make it plain that the issue is about the aiyprojects branch.

**Comment by blandiant** [comment]
@PaulFinch Hello, How did you manage to use other languages (on voicekit) ? (i  can't do it...) Do I have to update the sd image?
just to know, what the difference between the voicekit branch and the aiyprojects branch?
Thank you

**Comment by PaulFinch** [comment]
The main diff is the google assistant library version, aiyprojects is 0.0.3 , the voicekit branch is 0.1.0

The 0.1.0 can handle other languages, as you can change the language in the assistant phone app.
That said, it didnt work out of box for me, i had to re-register the device with the google assistant sdk.

**Comment by blandiant** [comment]
thank you

**Comment by tei-chino** [comment]
Dear Sir,
I downloaded the voice branch zip file, but I found that the files remained as old version. For example, the requirements.txt is still being "google-assistant-grpc==0.0.2", rather than "google-assistant-grpc==0.1.0" .  would you like to check the download file to see if there is any problem.

**Comment by drigz** [comment]
@tei-chino Make sure you're downloading the latest version of the repository. You may want to use `git clone` instead of downloading a zip.

**Comment by tei-chino** [comment]
Thanks for your advice.
However, I tried to use  
git clone https://github.com/google/aiyprojects-raspbian.git AIY-projects-python,
but the files still are old ones ( "google-assistant-grpc==0.0.2").

**Comment by drigz** [comment]
Only the `voicekit` branch has been updated so far - the `aiyprojects` branch will be updated soon.

**Comment by tei-chino** [comment]
Thanks for your information. 
Would you like tell me when the the image file in the voice kie site
https://aiyprojects.withgoogle.com/voice
will be updated?

**Comment by drigz** [comment]
We're working on it, but we don't have a date yet. We'll update this issue once it has been released.

**Comment by tei-chino** [comment]
Thanks for your rapid response. We are looking forword to the updated image.

**Comment by tei-chino** [comment]
I think that the new update allow us to interact using languages other than english, not just TTS.
I checked the source file device_helpers.py, but I could not find any language settings for my device.  Would you like to tell me how we can switch the laungauge for Assistant dialogue.

**Comment by drigz** [comment]
@tei-chino You have to change the language on your phone. See this issue for more details: https://github.com/googlesamples/assistant-sdk-python/issues/150

**Comment by tei-chino** [comment]
Thanks for your information.
I found the following information.
https://stackoverflow.com/questions/48017507/how-do-i-change-the-language-locale-in-the-google-assistant-sdk
I understood that becuase the AIY use the Google Assistant Library , we have to change the language in the Google Assistant settings on the phone. It is not possible to change language programmatically

**Comment by Tetsuo30** [comment]
Hi , I understand that for now the only solution is to change the language in the APP on smartphone ...
I own a galaxy A3 2016 (this smartphone has the requierment for the assistant APP) but the app isn't avalaible for it.  Is there any way to run the assistant app on this phone, or in emulator ? or no news about change the language programmatically ?

**Comment by drigz** [comment]
@Tetsuo30 If you're using assistant_grpc_demo.py or cloudspeech_demo.py you can change the language programmatically, although you won't have hotwording. https://github.com/googlesamples/assistant-sdk-python/issues/150 tracks programmatically changing the language with the Assistant Library.

**Comment by tei-chino** [comment]
@Tetsuo30 If you upgrade your smartphone's OS to Android 6.0, maybe you can install the Assistant App to your smartphone.

**Comment by blandiant** [comment]
Hi, before opening a new issue, 
I want to know if you have a solution: 
I change the language to french and now, to activate the Google assistant, i have to speak at 10cm from the mic or to yell...

**Comment by drigz** [comment]
@blandiant That sounds like an issue in the Assistant Library. If it hasn't already been reported here: https://github.com/googlesamples/assistant-sdk-python/issues then that is the place to report it.

**Comment by Tetsuo30** [comment]
Sorry but when I try to change language with the assistant_grpc_demo.py, it doesn't work . 
I try this in the terminal : assistant_grpc_demo.py -- lang fr-FR but no effect ... maybe I'm wrong , but can't find how to do this ...

**Comment by drigz** [comment]
@Tetsuo30 You need to add `aiy.i18n.set_language_code('fr-FR')` to the demo file.

**Comment by Tetsuo30** [comment]
Yes thx, I tried this code in the assistant_library_demo.py but forgot to try it in the grpc_demo.py.  It works,. But now, when asking something ...the sound stop in the middle of the response ... after 3 or 4 seconds ...

**Comment by lhvo** [comment]
I have the same pb as Tetsuo30.
Putting the locale code 'fr-FR' is working for recognizing but for the response it's truncated.
Removing the regional setting back to EN then everything works fine. 
Dumping the audio data, at the end thre was a lot of "0x00". Some limitation from google ?

**Comment by Tetsuo30** [comment]
Hi Ihvo, try the code from [Protovist](https://github.com/google/aiyprojects-raspbian/issues/297) in this post...it works for me ... The only issue for me now, is that I can't adjust the volume with the GRPC demo ... Impossible with the GUI or with the voice... when asking what's the volume the answer is always 50%, and can't lower it

**Comment by lhvo** [comment]
Greatt - confirmed that it's no longer truncated.
